openvm/tests/optimizer.rs

```diff
@@ -32,16 +33,16 @@ fn test_optimize() {
 
     println!(
         "Columns: {}, bus interactions: {}, constraints: {}",
-        machine.constraint_columns().len(),
+        machine.unique_columns().count(),
         machine.bus_interactions.len(),
         machine.constraints.len()
     );
     assert_eq!(
         [
-            machine.constraint_columns().len(),
+            machine.unique_columns().count(),
             machine.bus_interactions.len(),
             machine.constraints.len()
         ],
-        [562, 3207, 506]
+        [3540, 3207, 506]
```

Comment:
OK, this is one less than in the [other snapshot test](https://github.com/powdr-labs/powdr/blob/main/openvm/src/lib.rs#L717). I think this is because the final pre-compile has an additional `is_valid` column, that is not added here yet. But I think this deserves a comment, because people will have to update both tests in sync when changing the optimizer, and this is not obvious.